optee-utee/optee-utee-sys/src/user_ta_header.rs

```diff
@@ -16,7 +17,18 @@ pub struct ta_head {
     pub uuid: TEE_UUID,
     pub stack_size: u32,
     pub flags: u32,
-    pub entry: unsafe extern "C" fn(c_ulong, c_ulong, *mut utee_params, c_ulong),
+    pub depr_entry: u64,
+}
+
+extern "C" {
+    pub fn __utee_entry(func: c_ulong, session_id: c_ulong, up: *mut utee_params, cmd_id: c_ulong) -> TEE_Result;
+}
+
+#[no_mangle]
+pub fn __ta_entry(func: c_ulong, session_id: c_ulong, up: *mut utee_params, cmd_id: c_ulong) {
```

Comment:
`__ta_entry` is a diverging function. Perhaps, this line needs to change to:
```
pub fn __ta_entry(func: c_ulong, session_id: c_ulong, up: *mut utee_params, cmd_id: c_ulong) -> ! {
```

Reference: https://github.com/OP-TEE/optee_os/blob/master/ta/arch/arm/user_ta_header.c#L43

Reply:
Thanks.I'll update this.